subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ExecuteActionsTaskExecuter.java

```diff
@@ -461,45 +461,53 @@ private void executeAction(String actionDisplayName, TaskInternal task, InputCha
         buildOperationExecutor.run(new RunnableBuildOperation() {
             @Override
             public BuildOperationDescriptor.Builder description() {
-                return BuildOperationDescriptor.displayName(actionDisplayName + " for " + task.getIdentityPath().getPath()).name(actionDisplayName);
+                return BuildOperationDescriptor
+                        .displayName(actionDisplayName + " for " + task.getIdentityPath().getPath())
+                        .name(actionDisplayName)
+                        .details(ExecuteTaskActionBuildOperationType.DETAILS_INSTANCE);
             }
 
             @Override
             public void run(BuildOperationContext context) {
-                BuildOperationRef currentOperation = buildOperationExecutor.getCurrentOperation();
-                Throwable actionFailure = null;
                 try {
-                    action.execute(task);
-                } catch (Throwable t) {
-                    actionFailure = t;
-                } finally {
-                    action.clearInputChanges();
-                }
+                    BuildOperationRef currentOperation = buildOperationExecutor.getCurrentOperation();
+                    Throwable actionFailure = null;
+                    try {
+                        action.execute(task);
+                    } catch (Throwable t) {
+                        actionFailure = t;
+                    } finally {
+                        action.clearInputChanges();
+                    }
 
-                try {
-                    asyncWorkTracker.waitForCompletion(currentOperation, hasMoreWork ? RELEASE_AND_REACQUIRE_PROJECT_LOCKS : RELEASE_PROJECT_LOCKS);
-                } catch (Throwable t) {
-                    List<Throwable> failures = Lists.newArrayList();
+                    try {
```

Comment:
Can we remove the formatting clutter from this PR. makes it way more readable.

Reply:
We add an outer `try-finally` clause in this method. Apart from that the only changes are in lines `506` and `510`. Unfortunately, the diff is quite noisy due to the extra indentations needed within the try-block. 

But I don't think there is much I could do about that.